### PR TITLE
[core][dev-launcher] refactor network interceptor on ios

### PR DESCRIPTION
**Pull request by Kudo**
# Why

since we are going to integrate the network interceptor into expo go, this pr tries to move some shared code into expo-modules-core, fixing some issues and adding unit tests.

# How

- move the interceptor protocol class and cdp logger into expo-modules-core
- introduce strong typed `CdpNetworkTypes`
- fix http redirection issues
- fix duplicated CDP events when integrated with reanimated (we should exclude reanimated from the `sendWrappedEventToAllPages`

# Test Plan

- add `ExpoRequestCdpLoggerSpec` unit tests
- ci passed
- test network interceptor on bare-expo

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-dev-launcher/ios/DevLauncherNetworkInterceptor.swift [hunk 1]
```diff
@@ -0,0 +1,130 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+import ExpoModulesCore
+import React
+
+#if DEBUG && EX_DEV_CLIENT_NETWORK_INSPECTOR
+
+@objc(EXDevLauncherNetworkInterceptor)
+public final class DevLauncherNetworkInterceptor: NSObject, ExpoRequestCdpLoggerDelegate {
+  private static var isHookSetuped = false
```
Comment (tsapeta) [inline packages/expo-dev-launcher/ios/DevLauncherNetworkInterceptor.swift:10]: Hmm I think "setuped" is an incorrect past form. Generally "setup" is not a verb but a noun, so `isHookSetUp` would be more correct but I would just use `isHookInstalled`.
Moreover, moving this var to the top-level scope might be a good idea too.

File: packages/expo-modules-core/ios/Swift/Networks/ExpoRequestCdpLogger.swift [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+import Foundation
+
+/**
+ The `ExpoNetworkInterceptorProtocolDelegate` implementation to dispatch CDP (Chrome DevTools Protocol) events
+ */
+@objc
+public final class ExpoRequestCdpLogger: NSObject, ExpoRequestInterceptorProtocolDelegate {
+  private var delegate: ExpoRequestCdpLoggerDelegate?
+  internal var dispatchQueue = DispatchQueue(label: "expo.requestCdpLogger")
```
Comment (tsapeta) [inline packages/expo-modules-core/ios/Swift/Networks/ExpoRequestCdpLogger.swift:11]: Shouldn't it be a static var (or even top-level)?

File: packages/expo-modules-core/ios/Swift/Networks/ExpoRequestLoggerProtocol.swift [hunk 3]
```diff
@@ -0,0 +1,183 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+import Foundation
+
+/**
+ A `URLSession` interceptor which passes network events to its delegate
+ */
+@objc
```
Comment (tsapeta) [inline packages/expo-modules-core/ios/Swift/Networks/ExpoRequestLoggerProtocol.swift:8]: We may need to explicitly set an Objective-C name (with `EX` or `Expo` prefix) so it gets versioned correctly

File: packages/expo-modules-core/ios/Swift/Networks/ExpoRequestCdpLogger.swift (hunk 2 above)
Comment (Kudo) [reply packages/expo-modules-core/ios/Swift/Networks/ExpoRequestCdpLogger.swift:11]: the whole `ExpoRequestCdpLogger` class is a singleton already. it looks like moving `dispatchQueue` to static is redundant. does that make sense to you?

**Review by tsapeta — APPROVED** [review]
Except things we talked about on Slack, everything else looks good to me 👍

**Comment by Kudo** [comment]
the update e2e on android is breaking on main as well. i'm going to merge this pr.

**Pull request merged**